prime-router/src/main/kotlin/azure/observability/event/AzureEventUtils.kt

```diff
@@ -34,4 +38,25 @@ object AzureEventUtils {
     fun getIdentifier(bundle: Bundle): MessageID {
         return MessageID(bundle.identifier?.value, bundle.identifier?.system)
     }
+
+    /**
+     * Returns a ReportSentEvent
+     */
+    fun createReportSentEvent(
+        receiver: Receiver,
+        reportFiles: List<ReportFile?>,
+        reportId: ReportId,
+        externalFilename: String,
+    ): ReportSentEvent {
+        val reportFile = if (reportFiles.size == 1) reportFiles.first() else null
+        return ReportSentEvent(
+            reportFile?.reportId,
+            reportId,
+            receiver.topic,
+            Sender.createFullName(reportFile?.sendingOrg, reportFile?.sendingOrgClient),
```

Comment:
Following on from above, I think this should be a list?

Also, I would be a fan of throwing an error (or at least logging one) if the reportFile is missing the sender info since that would mean the data is in an a broken state.